### ISSUE TITLE
[Failing Test]: Many tests failing with invalid JWT signature

### ISSUE THREAD

**Issue by kennknowles** [apache/beam#31848]
### What happened?

 - [beam_CleanUpPrebuiltSDKImages](https://github.com/apache/beam/actions/runs/9883127590/job/27297285729#logs)
 - [beam_CleanUpGCPResources](https://github.com/apache/beam/actions/runs/9891127377/job/27320891963#logs)

```
ERROR: (gcloud.container.images.list) There was a problem refreshing your current auth tokens: ('invalid_grant: Invalid JWT Signature.', ***'error': 'invalid_grant', 'error_description': 'Invalid JWT Signature.'***)
Please run:

> Task :beam-test-tools:removeStaleSDKContainerImages FAILED
  $ gcloud auth login

to obtain new credentials.

If you have already logged in with a different account, run:

  $ gcloud config set account ACCOUNT

to select an already authenticated account to use.

FAILURE: Build failed with an exception.
```

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by kennknowles** [comment]
I think you have a fix up, yes? So I am assigning

**Comment by Abacn** [comment]
Yes I was trying to fix it - https://github.com/apache/beam/actions/workflows/beam_CleanUpGCPResources.yml?query=branch%3Atryfixpythontest

but not successful as some other affected workflows. Current plan to fix build-wheel first (which is more urgent one as it is used to build rc) then come back to these two

**Comment by Abacn** [comment]
The "Invalid JWT" error is affecting many performance tests that uses GitHub secret to set secret values also.

**Comment by Abacn** [comment]
CleanUpGCPResource currently permared due to two issues

- Some resources was created with some other credential, that the service account GitHub Action used doesn't have permission to delete them

- The JWT signature error observed in the test

For the second one, this happens since July 9th, and it is probablistic:

![image](https://github.com/user-attachments/assets/5448fadf-67a8-4a44-ad24-7654acb46f52)

This suggests some roll out process is in play. Either GCP or GitHub is rolling out something that breaking the auth

**Comment by Abacn** [comment]
Current status:

[java_tests](https://github.com/apache/beam/actions/workflows/java_tests.yml?query=event%3Aschedule) and [python_tests](https://github.com/apache/beam/actions/workflows/python_tests.yml?query=event%3Aschedule) failing, other tests fixed after switching to self-hosted runner's default cred

**Comment by Abacn** [comment]
downgrade to P2 as this has been mitigated. Due to that SA key expiration now enforced in the testing project, even if we generate new key and request update it, the key expires in 90 days and will break again.
